app/services/SubmissionService.scala

```diff
@@ -54,6 +54,21 @@ class SubmissionService @Inject()(
       }
   }
 
+  def submit(pid: String, answers: AssociateUcrAnswers)(implicit hc: HeaderCarrier): Future[Unit] = {
+    val eori = answers.eori.getOrElse(throw ReturnToStartException)
+    val mucr = answers.mucrOptions.map(_.mucr).getOrElse(throw ReturnToStartException)
+    val ucr = answers.associateUcr.map(_.ucr).getOrElse(throw ReturnToStartException)
+
+    connector
+      .submit(AssociateUCRRequest(pid, eori, mucr, ucr))
```

Comment:
We should probably clear the answers *after* the submission is successful, that way we can resubmit if it fails

Reply:
As above, you will be redirected to the choice page, because cache will be empty. Disassociation doesn't work

Reply:
Now when you go through disassociation journey you'll not reach confirmation page, but choice page